test/cookbooks/test/recipes/source_centos7.rb

```diff
@@ -0,0 +1,11 @@
+node.default['yum']['centos-sclo']['enabled'] = true
+node.default['yum']['centos-sclo']['managed'] = true
+node.default['yum']['centos-sclo-rh']['enabled'] = true
+node.default['yum']['centos-sclo-rh']['managed'] = true
+include_recipe 'yum-centos'
+
+yum_package 'devtoolset-7-gcc-c++'
```

Comment:
This really should be included in the source recipe for the cookbook so that it works "out of the box" for users. We can use an attribute to turn this on/off for air gap environments.

Reply:
I agree that the package should be installed in mainline code.  But, I don't think this cookbook is the right place to set up a a yum repository.   We could set up repository access for centos, but that's a case where we have network access to the centos repository.  Building from source is probably a thing to do when network access is not available so setting up the repository seems pointless. 

We also are not setting up access to repositories for other platforms. RHEL setup is outside of this cookbook.   For Debian we are are running apt-get commands outside the cookbook to make it work.  Centos7 is, for me, an edge case that doesn't need special handling in the main recipe.

Reply:
We setup repositories in other cookbooks if we need them to get it working. I don't see why that can't be the case here, especially if we make it optional for air gapped environments. I think for most use cases, users would expect this to work out of the box without having to add more code to get it to work on EL7.